src/main/java/uk/gov/register/core/RegisterContext.java

```diff
@@ -17,22 +16,19 @@
 
 public class RegisterContext {
     private RegisterName registerName;
-
-    private RegistersConfiguration registersConfiguration;
-    private FieldsConfiguration fieldsConfiguration;
+    private ConfigManager configManager;
     private MemoizationStore memoizationStore;
     private DBI dbi;
     private Flyway flyway;
     private RegisterMetadata registerMetadata;
 
-    public RegisterContext(RegisterName registerName, RegistersConfiguration registersConfiguration, FieldsConfiguration fieldsConfiguration, MemoizationStore memoizationStore, DBI dbi, Flyway flyway) {
+    public RegisterContext(RegisterName registerName, ConfigManager configManager, MemoizationStore memoizationStore, DBI dbi, Flyway flyway) {
         this.registerName = registerName;
-        this.registersConfiguration = registersConfiguration;
-        this.fieldsConfiguration = fieldsConfiguration;
+        this.configManager = configManager;
         this.memoizationStore = memoizationStore;
         this.dbi = dbi;
         this.flyway = flyway;
-        this.registerMetadata = registersConfiguration.getRegisterMetadata(registerName);
+        this.registerMetadata = configManager.getRegistersConfiguration().getRegisterMetadata(registerName);
```

Comment:
I wonder if (in future) we could move the `getRegisterMetadata` method on to `ConfigManager`?